src/collection.js

```diff
@@ -121,4 +122,84 @@ Collection.pluck = function pluck (collection$, sinkProperty) {
     .startWith([]);
 };
 
+// convert a stream of items' sources snapshots into a stream of collections
+Collection.gather = function gather (itemsState$ , component, sources, handlers = {}, idAttribute = 'id') {
+  const makeDestroyable = component => (sources) => {
+    const sinks = component(sources);
+    return {
+      ...sinks,
+      remove$: xs.merge(sinks.remove$ || xs.never(), sources.destroy$)
+    };
+  };
+  const collection = Collection(makeDestroyable(component), sources, {
+    remove$: (collection, item) => collection.remove(item),
+    ...handlers
+  });
+  // each time a new item appears, it should be added to the collection
+  const addReducers$ = itemsState$
```

Comment:
This declaration is pretty long (60 lines, 5 operators). It makes it quite hard to read.

Can we extract each of the functions in the operators to a named function so it reads more like:

```js
const addReducers$ = itemsState$
  .fold(findNewItems, {prevIds: [], addedItems: []})
  .map(({addedItems}) => addedItems)
  .filter(addedItems => addedItems.length)
  .map(itemsToSourceStreams)
  .map(loadItems)
```

I think that will make it more clear what's going on.